packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -1019,6 +1041,45 @@ export class FluidDataStoreRuntime extends
             throw new LoggingError("Runtime is closed");
         }
     }
+
+    /**
+     * Summarizer client should not have local changes. These changes can become part of the summary and can break
+     * eventual consistency. For example, the next summary (say at ref seq# 100) may contain these changes whereas
+     * other clients that are up-to-date till seq# 100 may not have them yet.
+     */
+    private identifyLocalChangeInSummarizer(
```

Comment:
we are not very consistent how we do that check for DDSs vs. data store - in former case it's done by runtime class, in later case it's done by context.
Also, any change to reuse code between these two layers to reduce bundle size impact?

Reply:
Yeah, we aren't consistent because data store and DDS states are not consistent. For example, DDSes don't know if the client is summarizer or not. I could plumb all that state but the reason I kept it this way was to reduce too much churn since this is temporary and we will be disallowing local states in summarizer soon.